micall/utils/cigar_tools.py

```diff
@@ -0,0 +1,780 @@
+"""
+Module for handling CIGAR strings and related alignment formats.
+"""
+
+from math import ceil, floor
+import re
+from typing import Container, Tuple, Iterable, Optional, Set, Dict, List, Union
+from dataclasses import dataclass
+from functools import cached_property, reduce
+from itertools import chain, dropwhile
```

Comment:
I've tried adding .flake8 configuration here: https://github.com/cfe-lab/MiCall/pull/1037

Overall it doesn't seem to be a very good fit because in order to pass the project, we need to disable a lot of warnings.

Reply:
I have also run the flake8 with default configuration on the files added by this commit, and fixed all of its warnings. I hope this will solve the PyCharm errors as well, but I cannot confirm it on my machine.